sql/core/src/main/scala/org/apache/spark/sql/DataFrameNaFunctions.scala

```diff
@@ -468,12 +477,24 @@ final class DataFrameNaFunctions private[sql](df: DataFrame) {
       s"Unsupported value type ${v.getClass.getName} ($v).")
   }
 
+  private def toAttributes(cols: Seq[String]): Seq[Attribute] = {
+    cols.flatMap { colName =>
+      df.col(colName).expr.collect {
+        case a: Attribute => a
```

Comment:
is this corrected? If `df.col` returns a nested field `GetStructField("a", Attribute("b"))`, then we will return `Attribute("b")` which is unexpected.

Reply:
So this works, but we can make it clearer.
```
cols.map(name => df.col(name).expr).collect {
  case a: Attribute => a
}
```
Since we know that struct type column will be ignored later, we don't need to collect them here.

Reply:
You are absolutely right. Thanks for the suggestion!